modal/_asgi.py

```diff
@@ -17,45 +16,31 @@ async def fn(scope):
         assert function_call_id, "internal error: function_call_id not set in asgi_app() scope"
 
         # TODO: Add support for the ASGI lifecycle spec.
-        messages_from_app: asyncio.Queue[List[Dict[str, Any]]] = asyncio.Queue(1)
+        messages_from_app: asyncio.Queue[Dict[str, Any]] = asyncio.Queue(1)
```

Comment:
This doesn't need to be a `List[]` — and I think that's a bit confusing because the messages aren't sent atomically regardless due to being outputted from a generator.

Reply:
The output is sent individually from a generator, and this isn't for preventing partial responses but rather for preventing intermixed responses from the asgi app itself and our error messages, for example:

```python
async def handle_first_input_timeout():
            if scope["type"] == "http":
                await messages_from_app.put({"type": "http.response.start", "status": 502})
                # <---- At this point the asgi app could be queuing another message on our queue, effectively breaking up the two "related" queue entries we are adding
                await messages_from_app.put(
                    {
                        "type": "http.response.body",
                        "body": b"Missing request, possibly due to cancellation or crash",
                    }
                )
```

Another option for preventing this would be to add some kind of simple locking mechanism (a simple started_sending=True bool should suffice?), preventing either the user app or our own error responses from being sent if the other has already started queuing responses.

Reply:
Do you have a reproduction of this? It doesn't really make sense to me that a user app could send a response before actually receiving a request.